### PR TITLE
fix: remove afterPass throttling

### PR DESCRIPTION
**Pull request by patrickhulce**
R: @brendankenny 

fixes https://github.com/GoogleChrome/lighthouse/pull/1868#pullrequestreview-28817344

### CODE REVIEW

**Comment by patrickhulce** [comment]
@brendankenny does this address your concern from #1868?

File: lighthouse-core/gather/driver.js [hunk 1]
```diff
@@ -696,15 +696,15 @@ class Driver {
...
-      if (!flags.disableCpuThrottling) p.push(emulation.enableCPUThrottling(this));
-    } else {
-      p.push(emulation.disableNetworkThrottling(this));
-      p.push(emulation.disableCPUThrottling(this));
-    }
-    return Promise.all(p);
+    const useCpuThrottling = passConfig.useThrottling && !flags.disableCpuThrottling;
+    const useNetworkThrottling = passConfig.useThrottling && !flags.disableNetworkThrottling;
+    const cpuThrottleFunc = `${useCpuThrottling ? 'enable' : 'disable'}CPUThrottling`;
+    const networkThrottleFunc = `${useNetworkThrottling ? 'enable' : 'disable'}NetworkThrottling`;
```
Comment (brendankenny) [inline lighthouse-core/gather/driver.js]: string construction is pretty bad, but so was the original :P

Short of a refactor of `emulation.js`, what about
```js
setThrottlings(flags, passConfig) {
  const throttleCpu = passConfig.useThrottling && !flags.disableCpuThrottling;
  const cpuPromise = throttleCpu ? emulation.enableCPUThrottling(this) :
      emulation.disableCPUThrottling(this);

  const throttleNetwork = passConfig.useThrottling && !flags.disableNetworkThrottling;
  const networkPromise = throttleNetwork ? emulation.enableNetworkThrottling(this) :
      emulation.disableCPUThrottling(this);

  return Promise.all([cpuPromise, networkPromise]);
}
```

File: lighthouse-core/gather/gather-runner.js [hunk 2]
```diff
@@ -244,6 +244,9 @@ class GatherRunner {
       log.verbose('statusEnd', status);
     });
 
+    // Disable throttling so the afterPass analysis isn't throttled
+    pass = pass.then(_ => driver.setThrottling(options.flags, {useThrottling: false}));
```
Comment (brendankenny) [inline lighthouse-core/gather/gather-runner.js]: can you update the top outline to mention this as well? `endTrace (if requested) & endNetworkCollect & endThrottling` or something?

File: lighthouse-core/gather/driver.js (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-core/gather/driver.js]: sure

File: lighthouse-core/gather/gather-runner.js (hunk 2 above)
Comment (patrickhulce) [inline lighthouse-core/gather/gather-runner.js]: done

**Comment by brendankenny** [comment]
any reason appveyor keeps failing?

**Comment by paulirish** [comment]
image

**Comment by patrickhulce** [comment]
yes everything should be failing that grabs canary chrome, hence #1911

**Comment by brendankenny** [comment]
oh right, whoops, was thinking of the one that landed in 1.6.0

**Pull request merged**